src/builder.rs

```diff
@@ -489,7 +489,7 @@ fn append_dir_all(
     let mut stack = vec![(src_path.to_path_buf(), true, false)];
     while let Some((src, is_dir, is_symlink)) = stack.pop() {
         let dest = path.join(src.strip_prefix(&src_path).unwrap());
-        if is_dir {
+        if is_dir || is_symlink && follow && src.is_dir() {
```

Comment:
Could parentheses be used here around the added clause to explicitly show the precedence? Additionally could you add a comment as well saying that `is_dir` being false is about the symlink, and the call to `.is_dir()` reaches through the symlink?